File: tarea-3/etapa-1/Juguete.cpp
#include <iostream>
#include <string>
#include "../headers/Juguete.h"
#include "../headers/Mascota.h"

Juguete::Juguete(int i, std::string n) : Item(i, "juguete", n, -1) {}
void Juguete::usarItem(Mascota &mascota) {
	std::cout << "Usando " << tipo << " " << nombre << " en la Mascota" << std::endl;
	mascota.mostrarMascota();
} 

File: proyecto/Juguete.py
from Item import Item
from PyQt6.QtWidgets import QLabel
from PyQt6.QtGui import QPixmap

class Juguete(Item):
    def __init__(self, id, nombre, cantidad=-1):
        super().__init__(id, "Juguete", nombre, cantidad)

    def usar(self, mascota):
        if self._cantidad > 0:
            self._cantidad -= 1
        else:
            print(f"No hay más {self._nombre} disponibles.")
